dwave/optimization/include/dwave-optimization/nodes/mathematical.hpp

```diff
@@ -30,6 +30,11 @@ struct abs {
     constexpr T operator()(const T& x) const { return std::abs(x); }
 };
 
+template <class T>
+struct logical {
+    constexpr bool operator()(const T& x) const { return x; }
```

Comment:
Seems slightly confusing to me that this returns a `bool` instead of `T`. I guess we're relying on the implementation of the `UnaryOpNode` to cast this back to the data type for arrays, so it should be fine. But why not return type `T` here, to be sure some use of `auto` deep in the `UnaryOpNode` doesn't go awry?

Reply:
I guess this is a "problem" for `std::logical_not` as well. Obviously not a big deal, just thought I would note it.

Reply:
We keep the bool for a few reasons. The first is consistency with `std::logical_and`, `std::logical_not`, etc. The second reason is we [use this info for `.logical()`, `.max()` and `.min()`](https://github.com/arcondello/dwave-optimization/blob/e5eab5823cef206e1caa9064ce681091a27f73cb/dwave/optimization/src/nodes/mathematical.cpp#L935). And the third is that it's the simplest way to get a truth value.